user_management/src/user_management/JWTManager.py

```diff
@@ -28,36 +34,50 @@ def __init__(self, token_type, payload=None):
         self.token_type = token_type
         self.payload = payload
 
-    def decode_jwt(self, encoded_jwt):
+    def decode_jwt(self, encoded_jwt: str) -> (bool, dict, str):
         try:
             decoded_payload = jwt.decode(encoded_jwt, self.public_key, algorithms=[self.algorithm])
-            return True, decoded_payload
-        except:
-            return False, None
-
-    def generate_token(self, user_id):
-        now = datetime.utcnow()
-        expiration_time = now + timedelta(minutes=self.expire_minutes_reference)
-        payload = {
-            'user_id': user_id,
-            'exp': expiration_time,
-            'token_type': self.token_type
-        }
-        token = jwt.encode(payload, self.private_key, algorithm=self.algorithm)
-        return token
-
-    def is_authentic_and_valid_request(self, request):
-        encoded_jwt = request.headers.get('Authorization')
-        success, decoded_payload = self.decode_jwt(encoded_jwt)
+            return True, decoded_payload, None
+        except Exception as e:
+            return False, None, str(e)
+
+    def generate_token(self, user_id: int) -> (bool, str, str):
+        if not user_exist(user_id):
+            return False, None, 'User does not exist'
+        try:
+            now = datetime.utcnow()
+            expiration_time = now + timedelta(minutes=self.expire_minutes_reference)
+            payload = {
+                'user_id': user_id,
+                'exp': expiration_time,
+                'token_type': self.token_type
+            }
+            token = jwt.encode(payload, self.private_key, algorithm=self.algorithm)
+        except Exception as e:
+            return False, None, str(e)
+        return True, token, None
+
+    def is_authentic_and_valid_request(self, encoded_jwt: str) -> (bool, list, int):
+        success, decoded_payload, error_decode = self.decode_jwt(encoded_jwt)
+        errors = []
         if not success:
-            return False, 'Invalid token', None
-        if decoded_payload is None:
-            return False, 'Empty payload', None
-        if decoded_payload.get('exp') < datetime.now():
-            return False, 'Token expired', None
+            errors.append(error_decode)
+        elif decoded_payload is None:
+            errors.append('Token expired')
```

Comment:
Is this the right error?

Reply:
yes

Reply:
This part of the code was not updated, thanks for noticing